superset/config.py

```diff
@@ -585,6 +586,23 @@ class CeleryConfig(object):  # pylint: disable=too-few-public-methods
 # dictionary.
 JINJA_CONTEXT_ADDONS: Dict[str, Callable] = {}
 
+# A dictionary of macro template processors that gets merged into global
+# jinja template processors. The existing template processors get updated with this
+# dictionary, which means the existing keys get overwritten by the content of this dictionary.
+# The customized addons don't necessarily need to use jinjia templating language.
+# This allows you to define custom logic to process macro template.
+# The example below configures a custom presto template processor which implements its own logic
+# of processing macro template.
+# class CustomPrestoTemplateProcessor(PrestoTemplateProcessor):
+#     """Custom template processor for presto."""
+#     engine = "presto"
+#     def process_template(self, sql: str, **kwargs) -> str:
+#         """Processes a sql template."""
```

Comment:
I don't follow what is process_template doing in this case, could you add this in the comment & describe the use case, e.g. using $ rather that {{ }} and write the example you had in the test here as well
In addition to that explain if there will / could be collisions between the templates

Reply:
my initial thought was if it's good to share the $ regex parse to open source project, I think it's good.

"The existing template processors get updated with this dictionary, which means the existing keys get overwritten by the content of this dictionary." explains how key collisions are handled, I can elaborate more.